src/rust/aggregator/src/data/user_history.rs

```diff
@@ -395,7 +404,45 @@ async fn aggregate_change<'a>(
     order_id: &BigDecimal,
     market_id: &BigDecimal,
     time: &DateTime<Utc>,
+    txn_version: &BigDecimal,
+    event_idx: &BigDecimal,
 ) -> DataAggregationResult {
+    // Get some info
+    let record = sqlx::query!(
+        r#"
+            SELECT order_type as "order_type: OrderType", remaining_size
+            FROM aggregator.user_history
+            WHERE market_id = $1
+            AND order_id = $2
+        "#,
+        market_id,
+        order_id,
+    )
+    .fetch_one(tx as &mut PgConnection)
+    .await
+    .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
+    let (order_type, original_size): (OrderType, BigDecimal) =
+        (record.order_type, record.remaining_size);
+    // If its a limit order and needs reordering
+    if matches!(order_type, OrderType::Limit) && &original_size < new_size {
+        let txn = txn_version.to_bigint().ok_or(DataAggregationError::ProcessingError(anyhow!("txn_version not integer")))? << 64;
+        let event = event_idx.to_bigint().ok_or(DataAggregationError::ProcessingError(anyhow!("event_idx not integer")))?;
+        let txn_event: BigDecimal = BigDecimal::from(txn & event);
+        sqlx::query!(
+            r#"
+                UPDATE aggregator.user_history_limit
+                SET last_increase_stamp = $3
```

Comment:
Don't you need to use txn version and event idx here too?

Reply:
![image](https://github.com/econia-labs/econia/assets/67070716/1f58867d-7ac3-4235-b3f9-8e86dea2eeaf)

It is using txn/idx. The thing is called stamp. Maybe I misunderstand the meaning of stamp but for me I replaced the time in timestamp with txn/idx and just called it stamp